ssl/quic/quic_impl.c

```diff
@@ -1320,7 +1320,7 @@ int ossl_quic_get_event_timeout(SSL *s, struct timeval *tv, int *is_infinite)
     QCTX ctx;
     OSSL_TIME deadline = ossl_time_infinite();
 
-    if (!expect_quic_any(s, &ctx))
+    if (!expect_quic_cs(s, &ctx))
```

Comment:
Can `SSL_get_event_timeout()` be called on Listener objects? If so this will fail.

Reply:
But if so, it would otherwise immediately crash when listening `ctx.qc`.  So perhaps not.  Does it really make sense to have a listener "timeout"?  Plausibly, listeners are *forever*.

Reply:
Do you have a suggested remedy for `ossl_quic_get_event_timeout()`?  Currently the deadline is derived from `ctx.qc`
```c
1328     if (ctx.qc->started)
1329         deadline
1330             = ossl_quic_reactor_get_tick_deadline(ossl_quic_channel_get_reactor(ctx.qc->ch));
```

Is there a different potential source for the deadline value in the case of QUIC listener?
If not, it is not clear what the maximum delay should be.

Reply:
Probably in the listener case we need something like this:

```` C
if (ossl_quic_port_is_running(ctx.ql->port)) 
        deadline = ossl_quic_reactor_get_tick_deadline(ossl_quic_engine_get0_reactor(ctx.ql->engine));
````
The reactor is also available via `ossl_quic_obj_get0_reactor(ctx.obj)` which should work in all cases - but I think we probably still need connection/listener specific code to check for the started/running status??

We probably also need a similar check in the handle_events code.

Reply:
Would this be via `ossl_quic_port_get0_reactor()` on the associated `QUIC_PORT`?  And even then there is currently no `get_time` for a QUIC listener.  Should it always be `current time`? The listener struct lacks the callback and arg fields.

```c
  52 static OSSL_TIME get_time(QUIC_CONNECTION *qc)
  53 {
  54     if (qc->override_now_cb != NULL)
  55         return qc->override_now_cb(qc->override_now_cb_arg);
  56     else
  57         return ossl_time_now();
  58 }
```

Reply:
Hmmm....although this throws up all sorts of questions about whether the original fix (for checking `ctx.qc->started`) was correct....when talking about ticking/deadlines we are talking about ticking/deadline for the whole `QUIC_ENGINE` which includes all connections in the event processing domain. So checking whether a single connection is started doesn't make much sense.

Reply:
Ah....but in master there is a one-2-one mapping between engine, port and channel so its not a big deal there. It's not the case in the quic-server branch